Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -435,24 +437,31 @@ def _addSampleLogInfo(self, ws, linePosition):
 
     def _moveDetector(self, ws, linePosition):
         """Perform detector position correction for direct and reflected beams."""
-        if self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault:
-            return ws
         # Reflected beam calibration
+        if self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault and \
+           self.getProperty(Prop.LINE_POSITION_INPUT).isDefault:
+            return ws
         detectorMovedWSName = self._names.withSuffix('detectors_moved')
+        args = {
+            'InputWorkspace': ws,
+            'OutputWorkspace': detectorMovedWSName,
+            'EnableLogging': self._subalgLogging,
+            'DetectorComponentName': 'detector',
+            'PixelSize': common.pixelSize(self._instrumentName),
+            'DetectorCorrectionType': 'RotateAroundSample',
+            'DetectorFacesSample': True
+        }
         directLineWS = self.getProperty(Prop.DIRECT_LINE_WORKSPACE).value
-        directLinePosition = directLineWS.run().getLogData(common.SampleLogs.LINE_POSITION).value
-        detectorMovedWS = SpecularReflectionPositionCorrect(
-            InputWorkspace=ws,
-            OutputWorkspace=detectorMovedWSName,
-            EnableLogging=self._subalgLogging,
-            DetectorComponentName='detector',
-            PixelSize=common.pixelSize(self._instrumentName),
-            LinePosition=linePosition,
-            DirectLinePosition=directLinePosition,
-            DirectLineWorkspace=directLineWS,
-            DetectorCorrectionType='RotateAroundSample',
-            DetectorFacesSample=True
-        )
+        if not self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault:
+            args['DirectLineWorkspace'] = directLineWS
+            args['DirectLinePosition'] = directLineWS.run().getLogData(common.SampleLogs.LINE_POSITION).value
+        else:
+            if not self.getProperty(Prop.TWO_THETA).isDefault:
+                args['TwoTheta'] = self.getProperty(Prop.TWO_THETA).value
+            else:
+                args['TwoTheta'] = ws.run().getLogData(common.SampleLogs.TWO_THETA).value
+        args['LinePosition'] = linePosition
```

Comment:
This assignment can be incorporated in the construction of the `dict`, lines 446-453.

Reply:
:+1: